arcgis-ios-sdk-samples/Display information/Add graphics with renderer/AddGraphicsWithRendererViewController.swift

```diff
@@ -0,0 +1,171 @@
+        return curvedGraphicsOverlay
+    }
+    
+    /// Create a heart-shape geometry with Bezier and elliptic arc segments.
+    ///
+    /// - Parameters:
+    ///   - center: The center of the square that contains the heart shape.
+    ///   - sideLength: The side length of the square.
+    /// - Returns: A heart-shape geometry.
+    func makeHeartGeometry(center: AGSPoint, sideLength: Double) -> AGSGeometry? {
+        guard sideLength > 0 else { return nil }
+        let spatialReference = center.spatialReference
+        // The x and y coordinates to simplify the calculation.
+        let minX = center.x - 0.5 * sideLength
+        let minY = center.y - 0.5 * sideLength
+        // The radius of the arcs.
+        let arcRadius = sideLength * 0.25
+        
+        // Bottom left curve.
+        let leftCurveStart = AGSPoint(x: center.x, y: minY, spatialReference: spatialReference)
+        let leftCurveEnd = AGSPoint(x: minX, y: minY + 0.75 * sideLength, spatialReference: spatialReference)
+        let leftControlPoint1 = AGSPoint(x: center.x, y: minY + 0.25 * sideLength, spatialReference: spatialReference)
+        let leftControlPoint2 = AGSPoint(x: minX, y: center.y, spatialReference: spatialReference)
+        let leftCurve = AGSCubicBezierSegment(start: leftCurveStart, controlPoint1: leftControlPoint1, controlPoint2: leftControlPoint2, end: leftCurveEnd, spatialReference: spatialReference)!
+        
+        // Top left arc.
+        let leftArcCenter = AGSPoint(x: minX + 0.25 * sideLength, y: minY + 0.75 * sideLength, spatialReference: spatialReference)
+        let leftArc = AGSEllipticArcSegment.createCircularEllipticArc(withCenter: leftArcCenter, radius: arcRadius, startAngle: .pi, centralAngle: -.pi, spatialReference: spatialReference)!
+        
+        // Top right arc.
+        let rightArcCenter = AGSPoint(x: minX + 0.75 * sideLength, y: minY + 0.75 * sideLength, spatialReference: spatialReference)
+        let rightArc = AGSEllipticArcSegment.createCircularEllipticArc(withCenter: rightArcCenter, radius: arcRadius, startAngle: .pi, centralAngle: -.pi, spatialReference: spatialReference)!
+        
+        // Bottom right curve.
+        let rightCurveStart = AGSPoint(x: minX + sideLength, y: minY + 0.75 * sideLength, spatialReference: spatialReference)
+        let rightCurveEnd = leftCurveStart
+        let rightControlPoint1 = AGSPoint(x: minX + sideLength, y: center.y, spatialReference: spatialReference)
+        let rightControlPoint2 = leftControlPoint1
+        let rightCurve = AGSCubicBezierSegment(start: rightCurveStart, controlPoint1: rightControlPoint1, controlPoint2: rightControlPoint2, end: rightCurveEnd, spatialReference: spatialReference)!
+        
+        let heart = [leftCurve, leftArc, rightArc, rightCurve].reduce(AGSMutablePart(spatialReference: spatialReference)) { (part: AGSMutablePart, segment: AGSSegment) -> AGSMutablePart in
+            part.add(segment)
+            return part
+        }
```

Comment:
`reduce` is really suited more toward value types. For something like this, it seems a bit overkill. You might consider doing something a bit simpler:

```swift
let heart = AGSMutablePart(spatialReference: spatialReference)
[leftCurve, leftArc, rightArc, rightCurve].forEach { heart.add($0) }
```